### PR TITLE
Make keyword substitution discoverable in studio

### PR DESCRIPTION
**Pull request by kluo**
Adds tips to course updates, enrollment email, and editing HTML
components explaining how to use keyword substitutions.

@caesar2164

### CODE REVIEW

File: cms/static/js/base.js [hunk 1]
```diff
@@ -15,15 +15,6 @@ domReady(function() {
...
     $body.addClass('js');
 
-    // lean/simple modal
-    $('a[rel*=modal]').leanModal({
-        overlay: 0.80,
-        closeButton: '.action-modal-close'
-    });
-    $('a.action-modal-close').click(function(e) {
-        (e).preventDefault();
-    });
```
Comment (caesar2164) [inline cms/static/js/base.js]: Why remove this code?

File: cms/static/js/views/modals/edit_xblock.js [hunk 2]
```diff
@@ -38,6 +38,15 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "js/vie
...
                 this.render();
                 this.show();
+                if (this.options.modalType == "html") {
+                    this.$('.editor-modes a[rel="modal"]').attr('title', gettext('This link will open in a modal window')).leanModal({
+                        overlay: 0.50,
+                        closeButton: '.action-modal-close'
+                    });
+                    this.$('.action-modal-close').click(function(e) {
+                        (e).preventDefault();
+                    });
```
Comment (caesar2164) [inline cms/static/js/views/modals/edit_xblock.js]: does this code replace the code in `cms/static/js/base.js` 1:1?

File: cms/static/sass/_base.scss [hunk 3]
```diff
@@ -685,66 +685,77 @@ hr.divider {
...
-
-      .label {
-        @extend %cont-text-sr;
-      }
-
-      [class^="icon-"] {
-        @extend %t-action1;
-        color: $white;
-      }
+// lean/simple modal window
```
Comment (caesar2164) [inline cms/static/sass/_base.scss]: I know you want to outdent to make CSS better, but could you maybe do it in a separate commit so we can see what CSS ACTUALLY changed?

**Comment by caesar2164** [comment]
@kluo - commented on a few areas asking for some more info...

File: cms/djangoapps/contentstore/views/component.py [hunk 4]
```diff
@@ -210,6 +211,7 @@ def container_handler(request, usage_key_string):
                 'xblock_info': xblock_info,
                 'draft_preview_link': preview_lms_link,
                 'published_preview_link': lms_link,
+                'keywords_supported': [{'name':keyword, 'desc': value.desc} for keyword, value in keyword_substitution.KEYWORD_FUNCTION_MAP.iteritems()],
```
Comment (stvstnfrd) [inline cms/djangoapps/contentstore/views/component.py]: I know this convention is used throughout the platform, but I don't care for single-line comprehensions.

They always seem more difficult to read and hence more prone to bugs.

Could we split this up like:
```python
'keywords_supported': [
    {
        'name': keyword,
        'desc': value.desc,
    }
    for keyword, value in keyword_substitution.KEYWORD_FUNCTION_MAP.iteritems()
]
```

File: cms/djangoapps/contentstore/views/course.py [hunk 5]
```diff
@@ -792,6 +793,7 @@ def settings_handler(request, course_key_string):
                 'test_email_url': reverse_course_url('send_test_enrollment_email', course_key),
                 'default_pre_template': default_enroll_email_template_pre,
                 'default_post_template': default_enroll_email_template_post,
+                'keywords_supported': [{'name':keyword, 'desc': value.desc} for keyword, value in KEYWORD_FUNCTION_MAP.iteritems()],
```
Comment (stvstnfrd) [inline cms/djangoapps/contentstore/views/course.py]: Actually, since we're using this same comprehension multiple times, would it make sense to wrap it in a helper function, `get_keywords_supported`?

File: cms/static/js/views/modals/edit_xblock.js [hunk 6]
```diff
@@ -38,6 +38,15 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "js/vie
                 this.editOptions = options;
                 this.render();
                 this.show();
+                if (this.options.modalType == "html") {
```
Comment (stvstnfrd) [inline cms/static/js/views/modals/edit_xblock.js]: `===`

File: cms/static/js/views/modals/edit_xblock.js [hunk 7]
```diff
@@ -38,6 +38,15 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "js/vie
...
+                    this.$('.editor-modes a[rel="modal"]').attr('title', gettext('This link will open in a modal window')).leanModal({
+                        overlay: 0.50,
+                        closeButton: '.action-modal-close'
+                    });
+                    this.$('.action-modal-close').click(function(e) {
+                        (e).preventDefault();
```
Comment (stvstnfrd) [inline cms/static/js/views/modals/edit_xblock.js]: Superfluous parens superfluously?

File: cms/static/js/base.js (hunk 1 above)
Comment (kluo) [inline cms/static/js/base.js]: See https://github.com/edx/edx-platform/pull/7351

File: cms/static/js/views/modals/edit_xblock.js (hunk 2 above)
Comment (kluo) [inline cms/static/js/views/modals/edit_xblock.js]: It's not a replacement, it's separately needed for the dynamically loading edit HTML modal

File: cms/static/sass/_base.scss (hunk 3 above)
Comment (kluo) [inline cms/static/sass/_base.scss]: This is just git / github being wonky, the only thing I did was remove the body.js wrapper, everything inside is the same.

**Comment by kluo** [comment]
@stvstnfrd ready for second pass

File: cms/static/js/views/modals/edit_xblock.js [hunk 8]
```diff
@@ -38,6 +38,15 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "js/vie
                 this.editOptions = options;
                 this.render();
                 this.show();
+                if (this.options.modalType === "html") {
```
Comment (stvstnfrd) [inline cms/static/js/views/modals/edit_xblock.js]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/settings.html [hunk 9]
```diff
@@ -294,8 +294,14 @@ <h2 class="title-2">${_("Course Enrollment Email")}</h2>
             <span class="tip">${_("Settings for emails that students receive upon course enrollment")}</span>
           </header>
           <section class="group-settings">
-            <p><strong>By default, this feature is disabled.</strong> You can either use the default email content or create your own.<br />
-              Students enrolling right on the start date will receive the post-course-start email.<br />
+            <p>${_("{strong_start}By default, this feature is disabled.{strong_end} You can either use the default email content or create your own. "
```
Comment (stvstnfrd) [inline cms/templates/settings.html]: If we're already breaking these long lines, we could wrap these a bit sooner and make the lines a bit shorter, especially the second one here.

File: common/djangoapps/util/keyword_substitution.py [hunk 10]
```diff
@@ -50,6 +50,19 @@ def add_keyword_function_map(mapping):
     KEYWORD_FUNCTION_MAP.update(mapping)
 
 
+def get_keywords_supported():
```
Comment (stvstnfrd) [inline common/djangoapps/util/keyword_substitution.py]: thanks!

File: cms/static/sass/_base.scss (hunk 3 above)
Comment (stvstnfrd) [inline cms/static/sass/_base.scss]: Yeah, the issue is the corresponding indent change after removing the parent node.

Alternatively, you _could_  remove _just_ the opening tag and closing brace in one commit, then correct the indent in a subsequent commit.

File: cms/static/js/views/modals/edit_xblock.js [hunk 11]
```diff
@@ -9,7 +9,7 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "js/vie
         var EditXBlockModal = BaseModal.extend({
             events : {
                 "click .action-save": "save",
-                "click .action-modes a": "changeMode"
+                "click .action-modes .action-item a": "changeMode"
```
Comment (stvstnfrd) [inline cms/static/js/views/modals/edit_xblock.js]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/js/basic-modal.underscore [hunk 12]
```diff
@@ -9,6 +9,11 @@
             <div class="modal-header">
                 <h2 class="title modal-window-title"><%= title %></h2>
                 <ul class="editor-modes action-list action-modes">
+                <% if (type == "html") { %>
```
Comment (stvstnfrd) [inline cms/templates/js/basic-modal.underscore]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/keyword_sub_modal.html [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+<div id="keyword-sub-modal" class="content-modal">
+  <a href="" rel="view" class="action action-modal-close">
+    <i class="icon-remove-sign"></i>
+    <span class="label">${_("close modal")}</span>
+  </a>
+
+  <h3 class="title">
+    ${_("Keyword Substitution")}
+    <span class="sr">,
```
Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/keyword_sub_modal.html [hunk 14]
```diff
@@ -0,0 +1,31 @@
+<%! from django.utils.translation import ugettext as _ %>
+<%page args="keywords_supported, content_type"/>
+
+<div id="keyword-sub-modal" class="content-modal">
+  <a href="" rel="view" class="action action-modal-close">
+    <i class="icon-remove-sign"></i>
+    <span class="label">${_("close modal")}</span>
```
Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/keyword_sub_modal.html [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+  </a>
+
+  <h3 class="title">
+    ${_("Keyword Substitution")}
```
Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/keyword_sub_modal.html [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    <span class="sr">,
+      ## Translators: this text gives status on if the modal interface (a menu or piece of UI that takes the full focus of the screen) is open or not
+      ${_("window open")}
```
Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: For consistency, single quotes for non-interpolated strings.

File: cms/templates/keyword_sub_modal.html [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    </span>
+  </h3>
+  <hr/>
+
+  <p>
+    ${_("Copy and paste keywords on the left into your {content_type}. They will be replaced by the values on the right.").format(content_type=content_type)}
```
Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: For consistency, single quotes for non-interpolated strings.

Comment (stvstnfrd) [inline cms/templates/keyword_sub_modal.html]: ignore comment.

File: cms/templates/js/basic-modal.underscore [hunk 18]
```diff
@@ -9,6 +9,11 @@
...
+                    <span class="tip">
+                      <a rel="modal" href="#keyword-sub-modal"><%= gettext("Supports keyword substitutions") %></a>
```
Comment (stvstnfrd) [inline cms/templates/js/basic-modal.underscore]: For consistency, single quotes for non-interpolated strings.

**Comment by caesar2164** [comment]
ok, I think I'm :+1: when tests pass.

**Comment by kluo** [comment]
retest this please
